### PR TITLE
Artifactory and bintray publish

### LINKED ISSUE #384: Publish opentelemetry-auto jar to maven central
This is a less ambitious version of #135 😄.

See https://github.com/open-telemetry/opentelemetry-auto-instr-java/issues/370#issuecomment-624223374.

I think there's some discussion we need to resolve in #370 first though.

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #384 

Publish functionality is applied to 4 modules for now: 3 exporters and `java-agent` module which produces the final fat-jar for the agent.

Although Bintray publishing is configured here, it probably is not usable yet, because actual version to publish is not configured. This is a topic of separate discussion and PR.

Publish of snapshot builds to Artifactory should work and our usual master build CI job is expected to actually do it.

### CODE REVIEW

**Comment by iNikem** [comment]
I would like to have a feedback on this question: do we want to automatically create maintenance branches whenever we release new major/minor version?

Pros:

- Will allow for fix PRs from external contributors as target branch will already exist

Cons:

- Requires CI to have write permission in GitHub repo
- Potentially many empty branches

File: .circleci/config.yml [hunk 1]
```diff
@@ -360,19 +343,9 @@ workflows:
       - publish_tag:
           requires:
             - test_7
-            # - test_8
```
[the remark this answers is not in the document]
Comment (iNikem) [reply .circleci/config.yml:363]: I think that my new jobs should supersede old ones. New `release` job is fired only on manual tag, so presumably we will tag only commits with all tests passed :). But `patch_release` indeed needs pre-requirements...

**Comment by iNikem** [comment]
Why CI is not triggered for this PR? Any ideas? :)

**Comment by iNikem** [comment]
Anybody wants to approve that? Or raise any concerns that require fixing? @trask @tylerbenson @jkwatson

**Comment by trask** [comment]
So sorry, I thought this was waiting on bintray permissions, will finish reviewing now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -337,19 +330,9 @@ workflows:
       - publish_master:
           requires:
             - test_7
-            # - test_8
```
[the remark this answers is not in the document]
Comment (trask) [reply .circleci/config.yml:312]: did u mean to resolve this conversation?

[the remark this answers is not in the document]
Comment (trask) [reply .circleci/config.yml:312]: oh, i think i understand your comment below, so this `publish_master` is no longer used?

**Comment by iNikem** [comment]
> So sorry, I thought this was waiting on bintray permissions, will finish reviewing now.

Well, you are probably correct that merging this without proper credentials configured will only make a noise of failed builds. But Yay/Nay would still be nice :)

File: .circleci/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (iNikem) [reply .circleci/config.yml:312]: I am sorry, I am just being absentminded here. This job is certainly needed, it publishes snapshot builds from master branch. And it certainly must depend on `test_8`. Fixed

**Comment by iNikem** [comment]
@trask, @tylerbenson  I think this can now be merged. If somebody approves :)

**Pull request merged**